scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,221 @@
+
+
+# get the domain edomain path on the rhevm
+def get_edomain_path(api, edomain):
+    edomain_id = api.storagedomains.get(edomain).get_id()
+    edomain_conn = api.storagedomains.get(edomain).storageconnections.list()[0]
+    return (edomain_conn.get_path() + '/' + edomain_id,
+            edomain_conn.get_address())
+
+
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+    """Cleanup all the empty directories on the edomain/edomain_id/master/vms
+    else api calls will result in 400 Error with ovf not found,
+
+    Args:
+        api: API to chosen RHEVM provider.
+        edomain: Export domain of chosen RHEVM provider.
+        edomainip: edomainip to connect through ssh.
+        sshname: edomain ssh credentials.
+        sshpass: edomain ssh credentials.
+    """
+    print "RHEVM: Deleting the empty directories on edomain/vms file..."
+    ssh_client = make_ssh_client(edomainip, sshname, sshpass)
+    command = 'cd %s/master/vms && find . -type d -empty -delete' % path
+    exit_status, output = ssh_client.run_command(command)
+    if exit_status != 0:
+        print "RHEVM: Error while deleting the empty directories on path.."
+        print output
+
+
+def is_edomain_template_deleted(api, name, edomain):
+    """Checks for the templates delete status on edomain.
+
+    Args:
+        api: API for RHEVM.
+        name: template_name
+        edomain: Export domain of selected RHEVM provider.
+    """
+    try:
+        template = api.storagedomains.get(edomain).templates.get(name)
+        if not template:
+            return True
+        return False
+    except Exception as e:
+        logger.exception(e)
+
+
+def delete_edomain_templates(api, template, edomain):
+    """deletes the template on edomain.
+
+    Args:
+        api: API for RHEVM.
+        name: template_name
+        edomain: Export domain of selected RHEVM provider.
+    """
+    with lock:
+        creation_time = template.get_creation_time().strftime("%d %B-%Y")
+        name = template.get_name()
+        print 'Deleting {} created on {} ...'.format(name, creation_time)
+    try:
+        template.delete()
+        print 'waiting for {} to be deleted..'.format(name)
+        wait_for(is_edomain_template_deleted,
+                 [api, name, edomain], fail_condition=False, delay=5)
+        print "RHEVM: successfully deleted the template {}".format(name)
+    except Exception as e:
+        with lock:
+            logger.exception(e)
+
+
+def cleanup_templates(api, edomain):
+    templates = api.storagedomains.get(edomain).templates.list()
+    thread_queue = []
+    delete_templates = []
+    for template in templates:
+        delta = datetime.timedelta(days=3)
+        now = datetime.datetime.now().strftime('%Y-%m-%d')
+        template_creation_time = template.get_creation_time() + delta
+
+        if template.get_name().startswith('auto-tmp'):
+            if now > (template_creation_time + delta).strftime('%Y-%m-%d'):
```

Comment:
I wonder - can the strftimes be avoided? I understand the string comparison works because of the numbers, but still ...